Add documentation for Reference Table audit trail and change events (#14950)
DataDog/documentation: content/en/logs/guide/reference-tables.md

## Modify a Reference Table

> To modify an existing Reference Table with new data, select a table then click **Update Config** in the top right corner.

Reviewer: ```suggestion
To modify an existing Reference Table with new data, select a table and click **Update Config** on the top right corner.
```

If there is a Lookup Processor using a Reference Table for Log enrichment, then the enrichment stops. It may take up to 10 minutes for the enrichment to stop.

> ## Monitoring Reference Table Activity

Reviewer: ```suggestion
## Monitor Reference Table Activity
```

> Reference table activity can be monitored with [Audit Trail][2] or [Change Events][3]. To view the audit trail and change events for a specific reference table, select a table and click the cog dropdown in the top right next to **Update Config** (you will need org management permissions to view audit trail).

Reviewer: ```suggestion
You can monitor reference table activity with [Audit Trail][2] or [Change Events][3]. To view the audit trail and change events for a specific reference table, select the table and click the Settings icon next to **Update Config**.  You need org management permissions to view the audit trail.
```

### Audit Trail

> Audit trail for reference tables is used for tracking user triggered actions. Audit trail events are sent when a user initially uploads or imports a csv file, or creates, modifies, or deletes a reference table. The `reference_table_file` Asset Type will show import/upload events, and the `reference_table` Asset Type will show reference table events. The purpose of the audit trail is to provide observability into the content of a reference table.

Reviewer: ```suggestion
Use the audit trail for reference tables to track user-triggered actions. Audit trail events are sent when a user initially uploads or imports a CSV file, or when a user creates, modifies, or deletes a reference table. 

The `reference_table_file` Asset Type displays import/upload events and the `reference_table` Asset Type displays reference table events. The audit trail provides observability into the content of a reference table.
```

### Change Events

> Change events for reference tables are used for tracking either automated or user triggered actions. They are sent when a cloud file is imported either from a user or automatic refresh. While events track user triggered actions as well, they are mainly used for tracking imports triggered when Reference Tables automatically pull a new csv file. Events contain information for the success status, path, and table name of the import. If an error is encountered, information on the type of error is also provided.

Reviewer: ```suggestion
Use change events for reference tables to track automated or user-triggered actions. They are sent when a cloud file is imported from a user or automatic refresh. While events can track user-triggered actions, they are mainly used to track triggered imports when a reference table automatically pulls a new CSV file. 

Events contain information about the success status, path, and table name of the import. If an error occurs, information about the error type is provided.
```

### Alerting

> To be alerted on errors encountered during imports, reference table change events can be tracked with [Event Monitors][4]. Reference table change events are sent from the `reference_tables` source. Monitors can be created from the Monitors tab, but a pre-filled monitor can be created by clicking on the cog dropdown in the top right next to **New Reference Table +**.

Reviewer: ```suggestion
To be alerted on errors encountered during imports, use [Event Monitors][4] for reference table change events. Reference table change events are sent from the `reference_tables` source. 

You can create monitors from the **Monitors** tab, or click on the Settings icon next to **New Reference Table +** to generate a pre-filled monitor.
```
